Drv/Ip/test/ut/PortSelector.cpp

```diff
@@ -38,11 +38,7 @@ U16 get_free_port(bool udp) {
         ::close(socketFd);
         return 0;
     }
-    U16 port = address.sin_port;
-    // Check for root-only-port.  If so, recursively try again.
-    if (port < 1024) {
```

Comment:
We might want to keep the recursive call here, at least for one call, to be able to allocate two ports for the UDP tests.

Right now, there is still a chance to have the port1 and port2 in  [`Drv/Udp/test/ut/UdpTester.cpp`](https://github.com/nasa/fprime/blob/0fe467c6e21aad321a0f1c737576733aac582443/Drv/Udp/test/ut/UdpTester.cpp#L32) and (I guess) in [`Drv/Ip/test/ut/TestUdp.cpp`](https://github.com/nasa/fprime/blob/0e8a773234777575facdedc390419427df66abe2/Drv/Ip/test/ut/TestUdp.cpp#L20)

I managed to trigger it after running few thousands tests:

```
[==========] Running 4 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 2 tests from Nominal
[ RUN      ] Nominal.BasicMessaging
Setup to receive udp at 127.0.0.1:51661 and send to 127.0.0.1:39209
Setup to receive udp at 127.0.0.1:39209 and send to 127.0.0.1:51661
[       OK ] Nominal.BasicMessaging (0 ms)
[ RUN      ] Nominal.BasicReceiveThread
/home/fprime/Drv/Udp/test/ut/UdpTester.cpp:45: Failure
Expected: (port1) != (port2), actual: 53481 vs 53481
[  FAILED  ] Nominal.BasicReceiveThread (0 ms)
[----------] 2 tests from Nominal (1 ms total)
```

It can be solved in another PR

Reply:
I think I will see if I can fix it in this PR. I am tired of random failures.